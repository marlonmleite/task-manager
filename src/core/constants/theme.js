const theme = {
  colors: {
    primary: '#FF5900',
  },
}

export default theme
