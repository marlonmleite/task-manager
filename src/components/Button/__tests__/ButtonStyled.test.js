import React from 'react'
import { shallow } from 'enzyme'
import theme from 'core/constants/theme'
import ButtonStyled from '../styled'

describe('ButtonStyled', () => {
  it('should render', () => {
    const btn = shallow(<ButtonStyled theme={theme}>Ok</ButtonStyled>)

    expect(btn).toMatchSnapshot()
  })

  it('should have primary color', () => {
    const btn = shallow(<ButtonStyled theme={theme} color="primary">Ok</ButtonStyled>)

    expect(btn).toHaveStyleRule('background-color', theme.colors.primary)
  })
})
