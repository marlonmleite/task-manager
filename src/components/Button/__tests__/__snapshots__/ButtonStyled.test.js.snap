// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonStyled should render 1`] = `
.c0 {
  padding: 0.5rem 1rem;
  color: #fff;
  display: inline-block;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid;
  border-radius: 2px;
}

.c0:focus,
.c0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-image: linear-gradient(transparent,rgba(0,0,0,.05) 40%,rgba(0,0,0,.1));
}

.c0:focus {
  outline: none;
}

.c0.disabled,
.c0:disabled {
  opacity: .65;
  box-shadow: none;
}

.c0:not([disabled]):not(.disabled) {
  cursor: pointer;
}

<button
  className="c0"
>
  Ok
</button>
`;
